client/allocrunner/networking_cni.go

```diff
@@ -123,8 +158,202 @@ func (c *cniNetworkConfigurator) Setup(ctx context.Context, alloc *structs.Alloc
 		c.logger.Debug("received result from CNI", "result", string(resultJSON))
 	}
 
-	return c.cniToAllocNet(res)
+	allocNet, err := c.cniToAllocNet(res)
+	if err != nil {
+		return nil, err
+	}
 
+	// prepend the Consul DNS to the nameservers, if we have it; we don't need
+	// the port because the iptables rule redirects port 53 traffic to it
+	if tproxyArgs != nil && tproxyArgs.ConsulDNSIP != "" {
+		if allocNet.DNS == nil {
+			allocNet.DNS = &structs.DNSConfig{
+				Servers:  []string{},
+				Searches: []string{},
+				Options:  []string{},
+			}
+		}
+		allocNet.DNS.Servers = append([]string{tproxyArgs.ConsulDNSIP},
+			allocNet.DNS.Servers...)
+	}
+
+	return allocNet, nil
+}
+
+// setupTransparentProxyArgs returns a Consul SDK iptables configuration if the
+// allocation has a transparent_proxy block
+func (c *cniNetworkConfigurator) setupTransparentProxyArgs(alloc *structs.Allocation, spec *drivers.NetworkIsolationSpec, portMapping []cni.PortMapping, portLabels map[string]int) (*consulIPTables.Config, error) {
+
+	var tproxy *structs.ConsulTransparentProxy
+	var cluster string
+	var proxyUID string
+	var proxyInboundPort int
+	var proxyOutboundPort int
+
+	exposePorts := []string{}
+	outboundPorts := []string{}
```

Comment:
That's true but we're converting those labels to port numbers in this function. The reason it's `[]string` is because the `iptables.Config` (which comes from the Consul SDK) that we're ultimately passing this slice into takes a `[]string` for the `ExcludeInboundPorts` rule.